pandaharvester/harvesterpreparator/analysis_aux_preparator.py

```diff
@@ -50,31 +57,52 @@ def trigger_preparation(self, jobspec):
                     tmpLog.debug('getting via http from {0} to {1}'.format(url, accPath))
                     res = requests.get(url, timeout=180, verify=False)
                     if res.status_code == 200:
-                        with open(accPath, 'w') as f:
+                        tmpLog.debug('res.headers = {0}'.format(res.headers))
+                        # Should we check the returned size if too small through error?
+                        with open(accPath, 'wb') as f:
                             f.write(res.content)
-                        return_code = 0
+                        # if accPath file exists and zero size remove
+                        if os.path.exists(accPath) and os.path.getsize(accPath) == 0 :
+                            tmpLog.debug('Remove file - {0} size - {1}'.format(accPath,os.path.getsize(accPath)))
+                            os.remove(accPath)
+                        else:
+                            tmpLog.debug('Successfully fetched file - {0}'.format(accPath))
+                            return_code = 0
                     else:
                         errMsg = 'failed to get {0} with StatusCode={1} {2}'.format(url, res.status_code, res.text)
                         tmpLog.error(errMsg)
                 except requests.exceptions.ReadTimeout:
                     tmpLog.error('read timeout when getting data from {0}'.format(url))
                 except Exception:
+                    # if accPath file exists and zero size remove
+                    if os.path.exists(accPath) and os.path.getsize(accPath) == 0 :
+                        tmpLog.debug('Remove file - {0} size - {1}'.format(accPath,os.path.getsize(accPath)))
+                        os.remove(accPath)
                     core_utils.dump_error_message(tmpLog)
             elif url.startswith('docker'):
-                args = ['docker', 'save', '-o', accPath, url.split('://')[-1]]
-                try:
-                    tmpLog.debug('executing ' + ' '.join(args))
-                    p = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                    stdout, stderr = p.communicate()
-                    return_code = p.returncode
-                    if stdout is not None:
-                        stdout = stdout.replace('\n', ' ')
-                    if stderr is not None:
-                        stderr = stderr.replace('\n', ' ')
-                    tmpLog.debug("stdout: %s" % stdout)
-                    tmpLog.debug("stderr: %s" % stderr)
-                except Exception:
-                    core_utils.dump_error_message(tmpLog)
+                # test if docker command exists
+                # else test if singularity command exists
+                args = None
+                if shutil.which('docker') is not None:
```

Comment:
BTW this logic is not good to use singularity when both docker and singularity are available. I suggest adding a plugin config parameter like containerRuntime and have an if-else using the parameter.